vignettes/spatial_dhmv_query.Rmd

```diff
@@ -0,0 +1,397 @@
+-   We can also find that the Coordinate Reference System (CRS) `crsSupported` is EPSG:31370, aka. [Belgian Lambert 72](https://epsg.io/31370). Always good to know.
+-   Finally, the capabilities xml provides a `version`: 2.0.1 at the time of writing. Yet keep that in mind for a bit.
+
+The "capabilities" are, per OGC standards, your go-to place for finding API documentation.
+
+In R, we can receive further information conveniently via `ows4R`. This works via the "coverage summary" &#x2026;
+
+```{r}
+feature <- "DHMVII_DTM_1m"
+dtm_smry <- caps$findCoverageSummaryById(feature, exact = TRUE)
+dtm_smry
+```
+
+&#x2026; and the coverage description.
+
+```{r}
+# Get description from the WCS client
+dtm_des <- WCS$describeCoverage(feature)
+dtm_des
+```
+
+
+Finally, it is important to know which dimensions are available: `LON/LAT`, `x/y`, or maybe `time`?
+
+```{r}
+dtm_dims <- dtm_smry$getDimensions()
+dtm_dims
+```
+
+
+So far, so good.
+
+Now, following the tutorials to get some real data:
+
+```{r}
+x_test <- 148600
+y_test <- 208900
+radius <- 1 # (+/- m)
+
+hopo_boxed <- OWSUtils$toBBOX(
+  xmin = x_test - radius,
+  xmax = x_test + radius,
+  ymin = y_test - radius,
+  ymax = y_test + radius
+  )
+```
+
+I will write the file to disk:
+
+```{r, eval=FALSE}
+tryCatch({
+  dtm_data = dtm_smry$getCoverage(
+    bbox = hopo_boxed,
+    filename = tempfile(fileext = ".tif")
+    )
+  dtm_data
+}, error = function (err) {message(err)})
+```
+
+&#x2026; or, maybe not. The function fails, somehow.
+
+*Status quo:*
+
+-   I receive a `tif` file (because *I called it* "tif"), supposedly some kind of GeoTIFF image (because I *asked* for `image/tiff`).
+-   I also receive a couple of error messages and warnings:
+    -   *"Start tag expected, '<' not found"*
+    -   *"cannot open this file as SpatRaster:&#x2026;"*
+    -   *"not recognized as being in a supported file format. (GDAL error 4)"*
+
+Maybe most useful, `ows4R` briefly displays the URL it used to attempt data query: `https://geo.api.vlaanderen.be/DHMV/wcs?service=WCS&version=2.0.1&coverageId=DHMVII_DTM_1m&subset=x(148599,148601)&subset=y(208899,208901)&format=image/tiff&request=GetCoverage`
+
+Let's break this URL down:
+
+-   the desired `service` is indeed `WCS`,
+-   `version` is `2.0.1` (I would guess)
+-   `request`, `coverageID`, and `format` seem to be as expected
+-   there are `subset` components for the respective dimensions.
+
+Feel free to try yourself to paste the URL to a browser and download the file.
+
+Still, the downloaded file is unreadable. Subtle symptom: even if one adjusts the `subset` (i.e. `bbox` argument, above), the received tif does not change in size; the download is bbox-agnostic, so to speak.
+
+Opening the file with a text editor such as [vim](https://www.vim.org), we see an XML part, some generic binary stuff, some hints that this [should be a GML](https://en.wikipedia.org/wiki/Geography_Markup_Language).
+
```

Comment:
After the xml part, the geoTiff part starts. It is very tricky however to know exactly where this starts and ends. This is what I have attempted to do with unpack_mht() function. However, that function failed on the download from the URL that you discuss here. The reason is likely an extra white line between the xml and geotiff part... 🤔

Reply:
Thank you for figuring out!
I will test this in a few minutes.

Note, however, that I would be hesitant to introduce `unpack_mht()`-like parsing. More of a philosophical reasoning: if `sf` cannot handle the data format, then why should this convenience package of ours support it? The parser is bound to break (again), because upstream format is not standardized. 
Instead, I would tend to report to DV and ask to get their file formats up to standards. (Or, more generally, document how to write correct queries.) 
That written, I'm glad your parser exists, and will probably use it.

Reply:
I agree that the parser should in principle not be needed. I remember having reported this to DV developers, but the forum is no longer online unfortunately.